contracts/contracts/SwapRouter.sol

```diff
@@ -0,0 +1,249 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity =0.7.6;
+pragma abicoder v2;
+
+import "@uniswap/v3-core/contracts/libraries/SafeCast.sol";
+import "@uniswap/v3-core/contracts/libraries/TickMath.sol";
+import "@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol";
+
+import "@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol";
+import "@uniswap/v3-periphery/contracts/base/PeripheryImmutableState.sol";
+import "@uniswap/v3-periphery/contracts/base/PeripheryValidation.sol";
+import "@uniswap/v3-periphery/contracts/base/PeripheryPaymentsWithFee.sol";
+import "@uniswap/v3-periphery/contracts/base/Multicall.sol";
+import "@uniswap/v3-periphery/contracts/base/SelfPermit.sol";
+import "@uniswap/v3-periphery/contracts/libraries/Path.sol";
+import "@uniswap/v3-periphery/contracts/libraries/PoolAddress.sol";
+import "@uniswap/v3-periphery/contracts/libraries/CallbackValidation.sol";
+import "@uniswap/v3-periphery/contracts/interfaces/external/IWETH9.sol";
+
+/**
+ * @title Uniswap V3 Swap Router
+ * @notice Router for stateless execution of swaps against Uniswap V3
+ * @dev code sourced from https://github.com/Uniswap/uniswap-v3-periphery/blob/main/contracts/SwapRouter.sol
```

Comment:
Recommend using permalink in case the current `main` ever changes

```suggestion
 * @dev code sourced from https://github.com/Uniswap/uniswap-v3-periphery/blob/1f5b385675eee30a77ee10f4dfd6aa0aaa766026/contracts/SwapRouter.sol
```

Reply:
Since this is quick, did this for you in 5166c1e0d513569b12b42ce41595121e3bcfaaaf